validation-test/SILOptimizer/constant_folded_fp_operations.swift

```diff
@@ -0,0 +1,633 @@
+    ) -> String {
+        let funcName = [unoptPrefix, fpType.printable_name(), op.printable_name(), op1.printable_name(), op2.printable_name()].joined(separator: "_")
+
+        return """
+        // CHECK-SIL-LABEL: sil hidden [noinline] [Onone] @\(funcName)
+        // CHECK-SIL-NEXT: bb0:
+        // CHECK-SIL: {{.*}}fcmp{{.*}}
+        // CHECK-SIL: } // end sil function '\(funcName)' 
+        """
+    }
+
+    func resultComaparisonCheck(
+        fpType: FPType, 
+        op: FPOperation, 
+        op1: FPOperand, 
+        op2: FPOperand
+    ) -> String {
+        let resultVarName = ["var", fpType.printable_name(), op.printable_name(), op1.printable_name(), op2.printable_name()].joined(separator: "_")
+
+        return """
+        // COMPARE-RESULT: [\(resultVarName)] result equal? true
+        """
+    }
+
+    func generateOptimizedFuncDecls() {
+        for fpType in FPType.allCases {
+            for op in FPOperation.allCases {
+                for op1 in FPOperand.allCases {
+                    for op2 in FPOperand.allCases {
+                        generateFuncDeclWithCheckDirectives(fpType: fpType, op: op, op1: op1, op2: op2, isopt: true)
+                    }
+                }
+            }
+        }
+    }
+
+    func generateUnoptimizedFuncDecls() {
+        for fpType in FPType.allCases {
+            for op in FPOperation.allCases {
+                for op1 in FPOperand.allCases {
+                    for op2 in FPOperand.allCases {
+                        generateFuncDeclWithCheckDirectives(fpType: fpType, op: op, op1: op1, op2: op2, isopt: false)
+                    }
+                }
+            }
+        }
+    }
+
+    func generateComparisonFuncDecls() {
+        for fpType in FPType.allCases {
+            for op in FPOperation.allCases {
+                for op1 in FPOperand.allCases {
+                    for op2 in FPOperand.allCases {
+                        let comparisonFuncName = ["comparison", fpType.printable_name(), op.printable_name(), op1.printable_name(), op2.printable_name()].joined(separator: "_")
+                        let resultVarName = ["var", fpType.printable_name(), op.printable_name(), op1.printable_name(), op2.printable_name()].joined(separator: "_")
+                        let optFuncName = [optPrefix, fpType.printable_name(), op.printable_name(), op1.printable_name(), op2.printable_name()].joined(separator: "_")
+                        let unoptFuncName = [unoptPrefix, fpType.printable_name(), op.printable_name(), op1.printable_name(), op2.printable_name()].joined(separator: "_")
+                        
+                        print("""
+                        @inline(never) @_silgen_name("\(comparisonFuncName)") @_optimize(none)
+                        func \(comparisonFuncName)() {
+                            let \(resultVarName) = \(optFuncName)() == \(unoptFuncName)()
+                            print("[\(resultVarName)] result equal? \\(\(resultVarName))")
```

Comment:
You could just emit a `precondition(result, "message")` instead of FileChecking the output. But checking the output is also fine

Reply:
Nice! Changed.